libqtile/window.py

```diff
@@ -1081,39 +1082,19 @@ def toscreen(self, index=None):
                 raise CommandError('No such screen: %d' % index)
         self.togroup(screen.group.name)
 
-    def match(self, wname=None, wmclass=None, role=None):
+    def match(self, match):
```

Comment:
I already like this a lot, but in my opinion we could go further: Maybe it would be nice to move catching the exceptions into `Match.compare` itself. `Window.match` could then be deprecated, unless the ability to match things in both directions is useful for purposes I am overlooking right now.